### ISSUE TITLE
Improve accessibility of signup form

### ISSUE THREAD

**Issue by praiztech** [wagtail/wagtail.org#426]
Modifications should be made to the newsletter signup form to improve its accessibility and ease of use. For example, 
- When screen reader users navigate a page by h2 and h3, they land on the same signup form. This can be confusing since the signup form is the only content under both headings
- The form is lacking an autocomplete attribute even though it collects personal information about the user
- Semantic HTML element - the fieldset - has been used for styling and layout purpose where a generic element is recommended
